### PR TITLE
[CP-2425] Refactor: Extracted history properties (HistoryLocationState) from actions

### PR DESCRIPTION
**Pull request by dkarski**
JIRA Reference: [CP-2425]

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2425]: https://appnroll.atlassian.net/browse/CP-2425?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/core/hooks/use-discovery-redirect-effect.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      return
+    }
+
+    if (discoveryDeviceInProgress) {
+      return
+    }
+
+    if (initializationDeviceInProgress) {
+      return
+    }
```
Comment (mkurczewski) [inline libs/core/core/hooks/use-discovery-redirect-effect.ts:46]: 🟢 What about?
```TSX
    if (deviceListEmpty || activeDeviceSet || discoveryDeviceInProgress || initializationDeviceInProgress) {
      return
    }
```

File: libs/core/device-initialization/components/devices-initialization-modal-flows/api-device-initialization-modal-flow.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { Dispatch } from "Core/__deprecated__/renderer/store"
+import { setDeviceInitializationStatus } from "Core/device-initialization/actions/base.action"
+import { DeviceInitializationStatus } from "Core/device-initialization/reducers/device-initialization.interface"
+
+export const APIDeviceInitializationModalFlow: FunctionComponent = () => {
+  const dispatch = useDispatch<Dispatch>()
+  const history = useHistory()
+
+  useEffect(() => {
+    history.push(`generic/mc-overview`)
```
Comment (mkurczewski) [inline libs/core/device-initialization/components/devices-initialization-modal-flows/api-device-initialization-modal-flow.tsx:19]: Shouldn't we take this path based on features delivered by the device? WDYT @lkowalczyk87 ?

Comment (lkowalczyk87) [reply libs/core/device-initialization/components/devices-initialization-modal-flows/api-device-initialization-modal-flow.tsx:19]: This part of the code will be overwritten it the next sprint so we can leave it as it is

Comment (dkarski) [reply libs/core/device-initialization/components/devices-initialization-modal-flows/api-device-initialization-modal-flow.tsx:19]: thx

File: libs/core/core/hooks/use-discovery-redirect-effect.ts (hunk 1 above)
Comment (dkarski) [reply libs/core/core/hooks/use-discovery-redirect-effect.ts]: https://github.com/mudita/mudita-center/pull/1684/commits/46ad41e1f5a7b02bbc2919e4cc17c9b15eda577c

**Pull request merged**